libraries/botbuilder-core/src/teams/teamsSSOTokenExchangeMiddleware.ts

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import * as signInConstants from '../signInConstants';
+import { ExtendedUserTokenProvider } from '../extendedUserTokenProvider';
+import { Middleware } from '../middlewareSet';
+import { Storage, StoreItem } from '../storage';
+import { TurnContext } from '../turnContext';
+
+import {
+    ActivityTypes,
+    StatusCodes,
+    TokenExchangeInvokeRequest,
+    TokenExchangeInvokeResponse,
+    TokenResponse,
+} from 'botframework-schema';
+
+function getStorageKey(context: TurnContext): string {
+    const activity = context.activity;
+
+    const channelId = activity.channelId;
+    if (!channelId) {
+        throw new Error('invalid activity-missing channelId');
+    }
```

Comment:
In the `TeamsSSOTokenExchangeMiddleware`, does the value of the channelId need to be [`botframework-schema.Channels.Msteams`](https://github.com/microsoft/botbuilder-js/blob/ada50e897946277c46474911f0cd6bbd42fe8e3b/libraries/botframework-schema/src/index.ts#L2116) or should this middleware fire for any signin/tokenExchange?

Reply:
I ported this closely from .NET - @EricDahlvang ?

Reply:
Deferring to the other comment for discussion.